File: Dopamine.Controls/IconTextButton.vb
Public Class IconTextButton
    Inherits Button

#Region "Properties"
    Public Property Data As Geometry
        Get
            Return CType(GetValue(DataProperty), Geometry)
        End Get

        Set(ByVal value As Geometry)
            SetValue(DataProperty, value)
        End Set
    End Property
#End Region

#Region "Dependency Properties"
    Public Shared ReadOnly DataProperty As DependencyProperty = DependencyProperty.Register("Data", GetType(Geometry), GetType(IconTextButton), New PropertyMetadata(Nothing))
#End Region

#Region "Construction"
    Shared Sub New()
        'This OverrideMetadata call tells the system that this element wants to provide a style that is different than its base class.
        'This style is defined in themes\generic.xaml
        DefaultStyleKeyProperty.OverrideMetadata(GetType(IconTextButton), New FrameworkPropertyMetadata(GetType(IconTextButton)))
    End Sub
#End Region
End Class

File: Dopamine.Common/Controls/IconTextButton.cs
using System;
using System.Windows;
using System.Windows.Controls;

namespace Dopamine.Common.Controls
{
    public class IconTextButton : Button
    {
        #region Properties
        public string Glyph
        {
            get { return Convert.ToString(GetValue(GlyphProperty)); }

            set { SetValue(GlyphProperty, value); }
        }

        public double GlyphSize
        {
            get { return Convert.ToDouble(GetValue(GlyphSizeProperty)); }

            set { SetValue(GlyphSizeProperty, value); }
        }
        #endregion

        #region Dependency Properties
        public static readonly DependencyProperty GlyphProperty = DependencyProperty.Register("Glyph", typeof(string), typeof(IconTextButton), new PropertyMetadata(null));
        public static readonly DependencyProperty GlyphSizeProperty = DependencyProperty.Register("GlyphSize", typeof(double), typeof(IconTextButton), new PropertyMetadata(null));
        #endregion

        #region Construction
        static IconTextButton()
        {
            //This OverrideMetadata call tells the system that this element wants to provide a style that is different than its base class.
            //This style is defined in themes\generic.xaml
            DefaultStyleKeyProperty.OverrideMetadata(typeof(IconTextButton), new FrameworkPropertyMetadata(typeof(IconTextButton)));
        }
        #endregion
    }
}
